main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    each do |i|
+      yield i
+    end
+    self
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield i, index
+      index + 1
+    end
+  end
+
+  def my_select
+    return unless block_given?
+
+    result = []
+    my_each do |i|
+      result.push(i) if yield i
+    end
+    result
+  end
+
+  def my_all?(*val)
+    result = true
+    if !val[0].nil?
+      my_each { |i| result = false unless val[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false unless yield(i) }
+    end
+    result
+  end
```

Comment:
# my_all?  
- [ ] `my_all?` when a class is passed as an argument returns true if all of the collection is a member of such class::Integer

     Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)

       expected true
            got false

- [ ] `my_all?` when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric

      Failure/Error: expect(numbers.my_all?(Numeric)).to be numbers.all?(Numeric)

        expected true
             got false

- [ ] `my_all?` when a Regex is passed as an argument returns true if all of the collection matches the Regex

      Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)

        expected true
             got false